modules/fleet-app-operator-permissions/variables.tf

```diff
@@ -37,11 +37,18 @@ variable "groups" {
 }
 
 variable "role" {
-  description = "The principals role for the Fleet Scope (`VIEW`/`EDIT`/`ADMIN`)."
+  description = "The principal's predefined role for the Fleet Scope (`VIEW`/`EDIT`/`ADMIN`). Either a predefined role or a custom role should be set"
   type        = string
   validation {
     condition     = contains(["VIEW", "EDIT", "ADMIN"], var.role)
     error_message = "Allowed values for role are VIEW, EDIT, or ADMIN."
   }
+  default     = ""
+}
+
+variable "custom_role" {
+  description = "The principal's role for the Fleet Scope which is a custom Kubernetes ClusterRole. Either a predefined role or a custom role should be set"
+  type        = string
+  default     = ""
```

Comment:
Some of the above logic might be simpler if the default was `null`.  Many of the variables in this repo use `""` as that predate when that became available in TF.